KA-Flocking/Assets/Scripts/BehaviourScripts/FightOrFlightBehaviour.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[CreateAssetMenu(menuName = "Flock/Behaviour/FightOrFlight")]
+public class FightOrFlightBehaviour : FlockBehaviour
+{
+    Vector3 enemiesDirection = Vector3.zero;
+    Vector3 friendsDirection = Vector3.zero;
+
+    float enemiesStrength = 0f;
+    float friendsStrength = 0f;
```

Comment:
Why is enemiesDirection, friendsDirection, enemiesStrength, friendsStrength instantiated here when they are reset in CalculateDirection() ?
The only reason for these to be global variables is to communicate between a helper function and it's caller. This generally means that calculateDirections() Should return the 4 values instead of setting global variables. Since OO languages generally don't implement tuples properly I understand if you want to do this approach, but I still don't like it :( (Not requested in changes)

Reply:
Would also be cool if you found a way to remember the calculations between updates, such as if they were variables on the flock agent's unit and merely mutated in this function. Unsure how to implement though